Implement CLI / Config with Cobra / Viper (#1043)
ansible/receptor: docs/source/user_guide/tls.rst

Configuring TLS
---------------

> Add ``tls-servers`` and ``tls-clients`` definitions to Receptor config files.

Reviewer: ```suggestion
Add ``tls-servers`` and ``tls-clients`` definitions to Receptor configuration files.
```

``foo.yml``

.. code-block:: yaml

    ---
    version: 2
    node:
      id: foo

    log-level:
      level: Debug

    tls-servers:
      - name: myserver
        cert: /full/path/foo.crt
        key: /full/path/foo.key
        requireclientcert: true
        clientcas: /full/path/ca.crt

    tcp-listeners:
      - port: 2222
        tls: myserver

> Defining ``tls-servers`` takes no effect, but it can be referenced elsewhere in the Receptor config file.

Reviewer: ```suggestion
Defining ``tls-servers`` has no effect but can be referenced elsewhere in the Receptor configuration file.
```

In the preceding configuration snippet, ``tls`` in the ``tcp-listeners`` is set to use ``myserver``.
In general, ``tls-servers`` should be referenced anywhere Receptor is expecting an incoming connection, such as ``*-listeners`` backends or on the ``control-services``.

> Similarly, ``tls-clients`` should be referenced anywhere Receptor is expecting to make an outgoing connection, i.e. ``*-peers`` backends or in ``receptorctl`` (the command-line client for Receptor).

Reviewer: ```suggestion
Similarly, ``tls-clients`` should be referenced anywhere Receptor is expecting to make an outgoing connection, such as ``*-peers`` backends or in ``receptorctl`` (the command-line client for Receptor).
```

    done

The preceding script will create a CA, and for each node ``foo`` and ``bar``, create a certificate request and sign it with the CA.

> These certificates and keys can then create ``tls-servers`` and ``tls-clients`` definitions in the Receptor config files.

Reviewer: ```suggestion
These certificates and keys can then create ``tls-servers`` and ``tls-clients`` definitions in the Receptor configuration files.
```
